src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -1054,7 +1055,8 @@ bool clang_c_convertert::get_type(const clang::Type &the_type, typet &new_type)
     get_decl_name(rd, name, id);
 
     /* record in context if not already there */
-    get_struct_union_class(rd);
+    if (get_struct_union_class(rd))
+      return true;
```

Comment:
It's this change that's causing the 3x new unknowns. I don't think it is controversial: checking whether we could actually convert the declaration is not wrong, as opposed to ignoring that result.

It fails for this declaration:
```c

static const struct kparam_array __param_arr_gpio_tab = {
        .max = (sizeof(gpio_tab) / sizeof((gpio_tab)[0]) + (sizeof(struct {
                  int
                  : -!!(__builtin_types_compatible_p(
                            typeof((gpio_tab)), typeof(&(gpio_tab)[0])));
                }))),
        .num = ((void *)0),
        .ops = &param_ops_ushort,
        .elemsize = sizeof(gpio_tab[0]),
        .elem = gpio_tab};
```
specifically the declaration of the struct with the `__builtin_types_compatible_p` as part of the width of the bitfield.

I'll look into adding support for this expression type.

Reply:
Thanks for checking this, @fbrausse.